src/ir.rs

```diff
@@ -240,7 +247,7 @@ pub enum UnaryOp {
 #[derive(Debug, Clone, Hash, PartialEq, Eq)]
 pub enum Expr {
     // nullary
-    Const(BitVecValue),
+    Const(u128),
```

Comment:
I would prefer if you could keep this a `BitVecValue`. You can always start with a 128-bit BitVecValue and then later narrow down the width after inference.